### PR TITLE
Check class

### PR DESCRIPTION
**Pull request by koheiw**
As part of #2020, 

- consolidate `check_dots()` and `unused_dots()`
- rename `friendly_class_undefined_message()` to `check_class()`

There are changed in those functions to make them more useful.

### CODE REVIEW

**Comment by koheiw** [comment]
`convert()` raises annoying warnings because of the default value of `omit_empty`.

```
> convert(dfm("a b c"), "data.frame")
  doc_id a b c
1  text1 1 1 1
Warning message:
In convert.dfm(dfm("a b c"), "data.frame") :
  omit_empty not used for 'to = "data.frame"'
```

File: R/validator.R [hunk 1]
```diff
@@ -117,3 +117,40 @@ check_length <- function(x, min_len, max_len, arg) {
...
+#' quanteda:::check_class("tokens", "dfm_select")
+#' }
+check_class <- function(class, method) {
+    class_valid <- rownames(attr(utils::methods(method), "info"))
+    class_valid <- stringi::stri_replace_first_fixed(class_valid, paste0(method, "."), "")
+    class_valid <- class_valid[class_valid != "default"]
+    if (!length(intersect(class, class_valid)))
+        stop(method, "() only works on ", paste(class_valid, collapse = ", "), " objects.", call. = FALSE)
+}
+friendly_class_undefined_message <- check_class # for compatibility
```
Comment (kbenoit) [inline R/validator.R:138]: :+1:

but I will update the extension packages too to use the new functions

**Review by kbenoit — APPROVED** [review]
More great cleanup work!

Let’s add guidelines for the use of these functions by (quanteda) package developers to the Wiki.

I also can see in the individual changes list that now, codecov alerts lack of coverage for specific lines (which is great). There are few not covered, if you want to target them and add tests. 

Otherwise :+1: +1: +1:

**Comment by kbenoit** [comment]
On the `convert()` warning why not raise a separate issue?

There are many misunderstandings about this that appear on StackOverflow (for instance), but we cannot convert empty documents to many of the topic model formats. 
E.g. https://stackoverflow.com/a/65241434/4158274

**Comment by koheiw** [comment]
Opened separate issues #2030 and 2031.

**Pull request merged**